GVRf/Framework/jni/engine/renderer/renderer.cpp

```diff
@@ -57,6 +100,62 @@ int Renderer::getNumberTriangles(){
     return numberTriangles;
 }
 
+void swapFrameQueries() {
+    lastFrameQuery = currentFrameQuery;
+    currentFrameQuery = !currentFrameQuery;
+}
+
+void Renderer::startGpuTimer(int eye) {
+    if(!disjointTimerSupported) {
+        return;
+    }
+    glBeginQuery(GL_TIME_ELAPSED_EXT, queries[eye][currentFrameQuery]);
+}
+
+void Renderer::stopGpuTimer(int eye) {
+    if(!disjointTimerSupported) {
+        return;
+    }
+
+    glEndQuery(GL_TIME_ELAPSED_EXT);
+    if(eye == 0 && !monoscopic) {
+        return;
+    }
+
+    drawTime += getGpuTimerResult(0);
+    if(!monoscopic) {
+        drawTime += getGpuTimerResult(1);
+    }
+
+    swapFrameQueries();
+
+    frameCount++;
+    if((frameCount % 10) == 0) {
+        avgDrawTime = drawTime / 10.0f;
+        drawTime = 0.0f;
+    }
+}
+
+float Renderer::getGpuTimerResult(int eye) {
+    float gpuTime = 0.0f;
+    glGetQueryObjectuiv(queries[eye][lastFrameQuery], GL_QUERY_RESULT_AVAILABLE, &queriesAvailable);
+    if(queriesAvailable == 0) {
+        return gpuTime;
+    }
+
+    glGetIntegerv(GL_GPU_DISJOINT_EXT, &queriesDisjointOccurred);
+    if(!queriesDisjointOccurred) {
+        glGetQueryObjectui64vEXT__(queries[eye][lastFrameQuery], GL_QUERY_RESULT, &timeElapsed);
+        gpuTime = (float) ((double)timeElapsed / 1000000.0);
```

Comment:
ok, lollipop user-build shows 0ms.  must be an issue on lollipop then.  that sucks.  i can try switching to gpu timestamps rather than using time elapsed.